### ISSUE TITLE
Crash when using CTRL+D with `&sc`

### ISSUE THREAD

**Issue by dosisod** [uiua-lang/uiua#29]
Came across this crash when playing around with user input. A single file with `&sc` is enough to reproduce.

Running:

```
$ uiua run main.ua
The application panicked (crashed).
Message:  called `Option::unwrap()` on a `None` value
Location: /home/$USER/.cache/cargo/registry/src/index.crates.io-6f17d22bba15001f/uiua-0.0.4/src/sys.rs:551

Backtrace omitted.

Run with RUST_BACKTRACE=1 environment variable to display it.
Run with RUST_BACKTRACE=full to include source snippets.
Error: The interpreter has crashed!
Hooray! You found a bug!
Please report this at http://github.com/uiua-lang/uiua/issues/new

code:
main.ua:1:1
&sc

   ╭─[main.ua:1:1]
 1 │ &sc
   │ ───
───╯
```

**Comment by kaikalii** [comment]
That's odd. It works for me and someone who raised another issue I just looked at.
How are you running it?

**Comment by kaikalii** [comment]
I didn't know that sending EOF was something you could do.
I have a couple options to handle it.
I'm not sure what other languages do.
Should I throw an error or return empty string?
What is the intent of a user who does that?

**Comment by dosisod** [comment]
> Should I throw an error or return empty string?

I would say return an empty string IMO. If you compare it to reading a file, EOF just says "hey the file has finished reading", so to me it makes more sense to implicitly return an empty string as if you where reading an empty file.

Doing a quick and dirty survey of some REPLs from other languages, CTRL+D seems to be treated the same as CTRL+C (when the prompt is empty, more on that later). In the following examples, I used CTRL+D to quick each of these REPLs.

```
$ python3
Python 3.11.5 (main, Sep  2 2023, 14:16:33) [GCC 13.2.1 20230801] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>>

$ node
Welcome to Node.js v20.6.1.
Type ".help" for more information.
>

$ sqlite3
SQLite version 3.43.1 2023-09-11 12:01:27
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite>

$ bash
$
exit
```

Those were with no text in the prompt, but when you have text in the prompts, CTRL+D seems to be ignored (except in `bash`, which just exits).

> What is the intent of a user who does that?

For me, I usually just accidentally type CTRL+D instead of CTRL+C. Others might have more practical uses for using CTRL+D.

**Comment by kaikalii** [comment]
I would think that you'd want to be able to distinguish between the user pressing ENTER without any text and the user pressing CTRL+D.

**Comment by Zollerboy1** [comment]
> Others might have more practical uses for using CTRL+D.

> I would think that you'd want to be able to distinguish between the user pressing ENTER without any text and the user pressing CTRL+D.

Yes, definitely. CTRL+C sends a SIGINT to a program, which normally terminates the program immediately, while CTRL+D sends an EOF to the input of the program but the program can still continue to run.

For Uiua, I could imagine that `&sc` could be changed so that it just gives back an empty string when an EOF is encountered and that there is another specialized function that has two outputs, the first one is either 0 or 1 depending on if the EOF is reached, while the second one is a string, same as the output of `&sc`.

Another possible solution could be that `&sc` outputs a 0 instead of a string in case EOF is reached and people who want to handle CTRL+D could just use `type` before doing anything with the string returned from `&sc`.

**Comment by kaikalii** [comment]
>Another possible solution could be that &sc outputs a 0 instead of a string in case EOF is reached and people who want to handle CTRL+D could just use type before doing anything with the string returned from &sc.

I like this solution.
I have implemented it in d34569a0faf369dd70.

**Comment by Zollerboy1** [comment]
Should `&rs`, `&rb`, and `&ru` behave in the same way?
